### PR TITLE
german-word-type-uberlegen

### LINKED ISSUE #508: False positive "überlegen"
<p></p>

<p></p>

### PR DESCRIPTION
**Pull request by lnazarenko**
fixes https://github.com/witty-works/nlp_api/issues/508

### CODE REVIEW

**Comment by lnazarenko** [comment]
fix this [508](https://github.com/witty-works/nlp_api/issues/508) But case like "wir sind überlegen" needs transformer-based model from SpaCy to make the part-of-the-speech tags more precisely

**Comment by lnazarenko** [comment]
I added a list of possible part-of-the-speech-tags to recognise words as adjectives or adv. It helps to fix an issue with "herausgefordert" in "Wir lieben es, herausgefordert zu werden." As it was recognise as perfect partciple which is correct in such case

**Comment by lnazarenko** [comment]
It works with small model but with larger model it works more precisely, as well as larger model helps to recognise plural or singular forms much better

File: tests/test_demo_wordings_german/test_internal_email/output.json [hunk 1]
```diff
@@ -231,21 +231,6 @@
             "subcategory": "agentic",
             "text": "herausgefordert"
         },
-        {
```
Comment (lsmith77) [inline tests/test_demo_wordings_german/test_internal_email/output.json:234]: `lieb,a,inclusive,communal` but here `lieben` is a verb

File: app/main.py [hunk 2]
```diff
@@ -1639,7 +1653,7 @@ def style_word_analysis_de(
     patterns = [model[lang.lang].make_doc(text) for text in terms]
     matcher.add("TerminologyList", patterns)
 
-    for i in range(len(tokens))[1:-1]:
```
Comment (lsmith77) [inline app/main.py:1642]: why did we skip the first and last token?

**Review by lsmith77 — APPROVED** [review]

**Pull request merged**